### PR TITLE
feat(page): add draw page

### LINKED ISSUE #56: Show random draw page
Show summary
Start tournament

### PR DESCRIPTION
**Pull request by RobbertWolfs**
fix #56

### CODE REVIEW

**Comment by RobbertWolfs** [comment]
First we show a button, when clicked it randoms shuffle teams / players.

screen shot 2018-03-14 at 23 01 53
screen shot 2018-03-14 at 23 01 56

File: src/pages/random.vue [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    getRandomPlayer() {
+      return this.players.splice(
+        (this.players.length * Math.random()) | 0,
+        1
+      )[0];
+    },
+
+    getRandomClub() {
+      return this.clubs.splice((this.clubs.length * Math.random()) | 0, 1)[0];
+    },
```
Comment (RobbertWolfs) [inline src/pages/random.vue]: Needs to be updated in the store afterwards

File: src/pages/summary.vue [hunk 2]
```diff
@@ -0,0 +1,59 @@
+<template lang="pug">
```
Comment (RobbertWolfs) [inline src/pages/summary.vue]: Needs to be merged after summary PR #58

Comment (RobbertWolfs) [reply src/pages/summary.vue]: done

File: src/pages/random.vue [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+    players() {
+      return this.teams.map(t => t.player);
+    }
+  }
+};
+</script>
+
+<style lang="scss">
+.matches {
```
Comment (RobbertWolfs) [inline src/pages/random.vue]: Rewrite css

File: src/pages/random.vue [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    },
+
+    players() {
+      return this.teams.map(t => t.player);
+    }
+  }
+};
+</script>
+
+<style lang="scss">
```
Comment (RobbertWolfs) [inline src/pages/random.vue]: check scoped styles

**Comment by RobbertWolfs** [comment]
Things to do:

When random:
- Add animations??
- Save random teams/clubs to store
- Go to summary

When summary:
- Change "create tournament" to "start tournament"
- Generate berger script (underneat the water according to Peter C.)
- Save berger result as matches in store
- Go to tournament type page (league)


Berger script needs to return a list of matches, one match should have to following structure:

```
{
    home: {
        team: 'Juventus',
        player: 'Wart',
        score: 0,
        winner: undefined
    },
    away: {
        team: 'AS Roma',
        player: 'Robbert',
        score: 0,
        winner: undefined 
    },
    state: matchStates.NONE
}
```

File: src/pages/random.vue (hunk 4 above)
Comment (RobbertWolfs) [reply src/pages/random.vue]: at a later moment

File: src/pages/random.vue (hunk 1 above)
Comment (RobbertWolfs) [reply src/pages/random.vue]: Added to the todo list

File: src/pages/random.vue (hunk 3 above)
Comment (RobbertWolfs) [reply src/pages/random.vue]: added to the todolist

**Comment by RobbertWolfs** [comment]
pwvff

@WartClaes

**Comment by RobbertWolfs** [comment]
Save id when random

File: src/pages/random.vue [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    },
+
+    getRandomClub() {
+      return this.clubs.splice((this.clubs.length * Math.random()) | 0, 1)[0];
+    },
+
+    startDraw() {
+      this.draw = true;
+
+      const i = 0;
```
Comment (WartClaes) [inline src/pages/random.vue]: nodig?

File: src/pages/random.vue [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+// .match__team {
+//   font-size: 20px;
+//   font-weight: 300;
+// }
+
+// .match__player {
+//   font-size: 12px;
+//   line-height: 20px;
+//   opacity: 0.5;
+// }
```
Comment (WartClaes) [inline src/pages/random.vue]: comments weg

File: src/store/mutations.js [hunk 7]
```diff
@@ -53,5 +53,17 @@ export default {
       tournament.matches = matches;
     }
   },
+
+  randomizeTeams(state, { slug, newTeams }) {
+    const tournament = state.tournaments.find(t => t.slug === slug);
+
+    console.log('teams', newTeams)
```
Comment (WartClaes) [inline src/store/mutations.js]: log weg

File: src/pages/random.vue (hunk 5 above)
Comment (RobbertWolfs) [reply src/pages/random.vue]: done

File: src/pages/random.vue (hunk 6 above)
Comment (RobbertWolfs) [reply src/pages/random.vue]: removed

File: src/store/mutations.js (hunk 7 above)
Comment (RobbertWolfs) [reply src/store/mutations.js]: removed

**Pull request merged**